src/components/InternalAddress.vue

```diff
@@ -0,0 +1,160 @@
+<!--
+  - SPDX-FileCopyrightText: 2020 Nextcloud GmbH and Nextcloud contributors
+  - SPDX-License-Identifier: AGPL-3.0-or-later
+-->
+
+<template>
+	<div>
+		<div v-for="domain in sortedDomains"
+			:key="domain.address">
+			{{ domain.address }}
+			{{ senderType(domain.type) }}
+			<ButtonVue type="tertiary"
+				class="button"
+				:aria-label="t('mail','Remove')"
+				@click="removeInternalAddress(domain)">
+				{{ t('mail','Remove') }}
+			</ButtonVue>
+		</div>
+		<div v-for="email in sortedEmails"
+			:key="email.address">
+			{{ email.address }}
+			{{ t('mail', 'domain') }}
+			<ButtonVue type="tertiary"
+				class="button"
+				:aria-label="t('mail','Remove')"
+				@click="removeInternalAddress(email)">
+				{{ t('mail','Remove') }}
+			</ButtonVue>
+		</div>
+		<ButtonVue type="primary"
+			@click="openDialog = true">
+			<template #icon>
+				<IconAdd :size="20" />
+			</template>
+			{{ t('mail', 'Add internal address') }}
+		</ButtonVue>
+		<NcDialog :open.sync="openDialog"
+			:buttons="buttons"
+			:name="t('mail', 'Add internal address')"
+			@close="openDialog = false">
+			<NcTextField class="input" :label="t('mail', 'Add internal email or domain')" :value.sync="newAddress" />
```

Comment:
a tiny bit of text would also help this dialog. you could show examples, like how to actually set a domain as internal. is it `domain.tld` or `@domain.tld` or `*@domain.tld` a user has to enter? :)

Reply:
all of them work
If you add name@domain.tld it will be saved as an address 
if you add @domain.tld it will be stripped into domain.tld 
if you add domain.tld it will be saved as a domain an addresses ares gonna be checked against it